src/Microsoft.IdentityModel.Tokens/EventBasedLRUCache.cs

```diff
@@ -121,22 +117,26 @@ private int EventQueuePollingInterval
         /// <param name="removeExpiredValues">Whether or not to remove expired items.</param>
         /// <param name="removeExpiredValuesIntervalInSeconds">The period to wait to remove expired items, in seconds.</param>
         /// <param name="maintainLRU">Whether or not to maintain items in a LRU fashion, moving to front of list when accessed in the cache.</param>
+        /// <param name="compactIntervalInSeconds">The period to wait to compact items, in seconds.</param>
         internal EventBasedLRUCache(
             int capacity,
             TaskCreationOptions options = TaskCreationOptions.None,
             IEqualityComparer<TKey> comparer = null,
             bool removeExpiredValues = false,
             int removeExpiredValuesIntervalInSeconds = 300,
-            bool maintainLRU = false)
+            bool maintainLRU = false,
+            int compactIntervalInSeconds = 20)
```

Comment:
Does this need to be this frequent? For comparison, the expired item removal is every 5 minutes.

Reply:
@pmaytak This is not how often we compact or try and compact, but the amount of time that must pass before we will compact again.
We compact when it is detected the capacity has been exceeded.